### ISSUE TITLE
Doesn't work with local emulator

### ISSUE THREAD

**Issue by tallpants** [teifip/gcf-api-router#1]
The router seems to fail completely when used with the cloud functions local emulator. https://yarnpkg.com/en/package/@google-cloud/functions-emulator

The process of using the emulator is described here: https://cloud.google.com/functions/docs/emulator

Here's my terminal:

```
$ functions start
Starting Google Cloud Functions Emulator...
Google Cloud Functions Emulator STARTED
┌────────┬────────┬─────────┬────────────────────────────────────────────┐
│ Status │ Name   │ Trigger │ Resource                                   │
├────────┼────────┼─────────┼────────────────────────────────────────────┤
│ READY  │ status │ HTTP    │ http://localhost:8010/1/us-central1/status │
└────────┴────────┴─────────┴────────────────────────────────────────────┘

$ functions deploy status --trigger-http
Copying file://C:\Users\TALLPA~1\AppData\Local\Temp\us-central1-status-11344Y3fxYbpo6B5B.zip...
Waiting for operation to finish...done.
Deploying function......done.
Function status deployed.
┌────────────┬──────────────────────────────────────────────────────────────────────────────────────┐
│ Property   │ Value                                                                                │
├────────────┼──────────────────────────────────────────────────────────────────────────────────────┤
│ Name       │ status                                                                               │
├────────────┼──────────────────────────────────────────────────────────────────────────────────────┤
│ Trigger    │ HTTP                                                                                 │
├────────────┼──────────────────────────────────────────────────────────────────────────────────────┤
│ Resource   │ http://localhost:8010/1/us-central1/status                                           │
├────────────┼──────────────────────────────────────────────────────────────────────────────────────┤
│ Timeout    │ 60 seconds                                                                           │
├────────────┼──────────────────────────────────────────────────────────────────────────────────────┤
│ Local path │ D:\Projects\Personal\railtrack                                                       │
├────────────┼──────────────────────────────────────────────────────────────────────────────────────┤
│ Archive    │ file://C:\Users\TALLPA~1\AppData\Local\Temp\us-central1-status-11344Y3fxYbpo6B5B.zip │
└────────────┴──────────────────────────────────────────────────────────────────────────────────────┘
```

On visiting the URL specified, I get this JSON response:

```json
{  
   "stack":"TypeError: Cannot read property 'charAt' of undefined\n    at findRequestHandlers (D:\\Projects\\Personal\\railtrack\\node_modules\\gcf-api-router\\index.js:87:28)\n    at RequestHandler.onRequest (D:\\Projects\\Personal\\railtrack\\node_modules\\gcf-api-router\\index.js:12:27)\n    at app.use (C:\\Users\\tallpants\\AppData\\Local\\Yarn\\config\\global\\node_modules\\@google-cloud\\functions-emulator\\src\\supervisor\\worker.js:130:11)\n    at Layer.handle [as handle_request] (C:\\Users\\tallpants\\AppData\\Local\\Yarn\\config\\global\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\tallpants\\AppData\\Local\\Yarn\\config\\global\\node_modules\\express\\lib\\router\\index.js:317:13)\n    at C:\\Users\\tallpants\\AppData\\Local\\Yarn\\config\\global\\node_modules\\express\\lib\\router\\index.js:284:7\n    at Function.process_params (C:\\Users\\tallpants\\AppData\\Local\\Yarn\\config\\global\\node_modules\\express\\lib\\router\\index.js:335:12)\n    at next (C:\\Users\\tallpants\\AppData\\Local\\Yarn\\config\\global\\node_modules\\express\\lib\\router\\index.js:275:10)\n    at app.use (C:\\Users\\tallpants\\AppData\\Local\\Yarn\\config\\global\\node_modules\\@google-cloud\\functions-emulator\\src\\supervisor\\worker.js:102:7)\n    at Layer.handle [as handle_request] (C:\\Users\\tallpants\\AppData\\Local\\Yarn\\config\\global\\node_modules\\express\\lib\\router\\layer.js:95:5)",
   "message":"Cannot read property 'charAt' of undefined",
   "name":"TypeError"
}
```

But when actually deployed to Google Cloud, it works flawlessly.

**Comment by tallpants** [comment]
It seems like the `req.params` is empty for some reason in the `findRequestHandlers` function, but I don't know how to debug this -- putting `console.log`'s in the function seems pointless because it'll never actually log anywhere.

**Comment by teifip** [comment]
@tallpants Thanks for your finding! Probably the simulator does not handle `req.params` as the real environment. This should be easy to fix by using `req.path` instead. For example, change the code at line 87 into:
```
let path = req.path === null ? '/' : req.path;
```
Need to confirm that there are no surprises in the way the simulator handles `req.path`. Just give me time to install the simulator tomorrow, and I will push the updated code if it works.

**Comment by teifip** [comment]
Just published revision 1.0.1 on [npm](https://www.npmjs.com). Again, thanks @tallpants for reporting the issue.
